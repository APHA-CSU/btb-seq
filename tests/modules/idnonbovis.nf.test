nextflow_process {

    name "Test Process idNonBovis"
    script "main.nf"
    process "IDNONBOVIS"

    test("Should run Kraken and Bracken to classify input fastq files") {

        when {
            params {
                kraken2db = "/opt/Kraken2/test-db/"
                lowmem = ""
                rmInter = false
            }
            process {
                """
                input[0] = ["phix-m", file("${projectDir}/tests/data/phix_test/outcome.txt"),
                            file("${projectDir}/tests/data/phix_test/trimmed_1.fastq"),
                            file("${projectDir}/tests/data/phix_test/trimmed_2.fastq")]
                input[1] = params.kraken2db
                input[2] = params.lowmem
                input[3] = params.rmInter
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}