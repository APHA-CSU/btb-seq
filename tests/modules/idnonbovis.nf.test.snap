{
    "Should run Kraken and Bracken to classify input fastq files": {
        "content": [
            {
                "0": [
                    "phix-m_bovis.csv:md5,2d181b21051a51392da003389706eded"
                ],
                "1": [
                    [
                        "phix-m",
                        [
                            "phix-m_CheckRequired-S1_brackensort.tab:md5,e0902240b1c6a98c6dd5f16613759495",
                            "phix-m_CheckRequired_brackensort.tab:md5,26dbaaef1d704d8feef94d4e8daa01e3"
                        ],
                        "phix-m_CheckRequired_kraken2.tab:md5,b47b5b7d1a5dfed118889d533b35840c"
                    ]
                ],
                "krakenOut": [
                    [
                        "phix-m",
                        [
                            "phix-m_CheckRequired-S1_brackensort.tab:md5,e0902240b1c6a98c6dd5f16613759495",
                            "phix-m_CheckRequired_brackensort.tab:md5,26dbaaef1d704d8feef94d4e8daa01e3"
                        ],
                        "phix-m_CheckRequired_kraken2.tab:md5,b47b5b7d1a5dfed118889d533b35840c"
                    ]
                ],
                "queryBovis": [
                    "phix-m_bovis.csv:md5,2d181b21051a51392da003389706eded"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-12T15:34:24.286004178"
    }
}